students/hirot_directory/session05/mailroom5.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+"""
+Mailroom5.py
+Dictionary for the menu selections
+Comprehension
+Exception Handling
+
+"""
+
+import sys
+
+donor_names = ["William Gates III", "Mark Zuckerberg", "Jeff Bezos", "Paul Allen"]
+donations = [[200, 500], [2000, 3000], [4000, 5000], [100, 150]]
+
+donor_dict = { donor_names:donations for (donor_names, donations) in zip(donor_names, donations)}
+
+
+def print_list():
+    # print(donor_dict)
+    print("{msg1: <50}| {msg2: <12}| " \
+
+          "{msg3:<10}| {msg4: <12}".format(msg1="Donor Name",
+
+                                           msg2="Total Given",
+
+                                           msg3="Num Gifts",
+
+                                           msg4="Average Gift"))
+    for d in donor_dict:
+        t = sum(donor_dict[d])
+        n = len(donor_dict[d])  # list.count() is to count inside of the list (i.e. # by blue)
+        a = t / n
+
+        print("{d: <50} ${t: 12.2f}{n: 12d}{a: 14.2f}".format(d=d, t=t, n=n, a=a))
+
```

Comment:
Looking good! See if you can figure out how to print them in order of total contributions.